### PR TITLE
Refactor long functions into smaller, descriptive functions 

### PR DESCRIPTION
**Pull request by aludvik**
This PR aims to improve readability by refactoring long functions into multiple, smaller, descriptively named functions.

Sorry @knkski I messed up on #20 and accidentally pushed to the bitwiseio fork instead of mine. Had to close that and open this, but I will address your comments here.

### CODE REVIEW

File: src/node.rs [hunk 1]
```diff
@@ -329,6 +194,156 @@ impl PbftNode {
...
+        debug!(
+            "{}: Received Checkpoint message from {:02}",
+            self.state,
+            self.state
+                .get_node_id_from_bytes(pbft_message.get_info().get_signer_id())?
+        );
+
+        if self.msg_log.get_latest_checkpoint() >= pbft_message.get_info().get_seq_num() {
+            debug!(
+                "{}: Already at a stable checkpoint with this sequence number or past it!",
```
Comment (agunde406) [inline src/node.rs:259]: remove the "!"

**Review by agunde406 — APPROVED** [review]
Tested locally

**Comment by aludvik** [comment]
@knkski in response to the below comment from #20, I would prefer to use a generic function over a macro, as I think they are easier to read.

> In the dynamic networking branch, this has been changed to Duration::from_secs (I guess since it got bumped up 30k ms). With that in mind, would it make sense to have something like this?
> 
> macro_rules! parse {
>     ($prop:expr, $name:expr) => {
>         $prop = settings.get($name)
>             .and_then(|s| s.parse().ok())
>             .unwrap_or($prop);
>     };
>     ($prop:expr, $name:expr, $convert:expr) => {
>         $prop = settings.get($name)
>             .and_then(|s| s.parse().ok())
>             .and_then(|d| Some($convert(d)))
>             .unwrap_or($prop);
>     };
> }
> 
> parse!(config.block_duration, "sawtooth.consensus.pbft.block_duration", Duration::from_millis);
> parse!(config.view_change_timeout, "sawtooth.consensus.pbft.view_change_timeout", Duration::from_secs);
> parse!(config.max_log_size, "sawtooth.consensus.pbft.max_log_size");
> 
>

**Comment by aludvik** [comment]
@knkski in response to the below comment, that would make sense, but I refactored the above type checking function in a later commit, which is why I didn't do it here.

> Should this go above the above line, since that has pbft_message.get_info() as well?

**Comment by aludvik** [comment]
@knkski in response to the below comment, I don't have a problem with additional refactoring as time allows, but that seems like a bigger change than I have time for at the moment.

> It looks like handlers.rs is only used by node.rs. Given that we're refactoring out bits of node.rs into smaller functions over in that file, should we just uproot and move handlers.rs entirely, or move all of the refactored smaller functions over here?

**Comment by aludvik** [comment]
Pushed a commit to make setting merger more generic

**Review by knkski — APPROVED** [review]
👍

**Pull request closed without merging**